src/main/java/frc/robot/commands/TagPositionDrive.java

```diff
@@ -0,0 +1,117 @@
+// Open Source Software; you can modify and/or share it under the terms of
+// the WPILib BSD license file in the root directory of this project.
+
+package frc.robot.commands;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+
+import com.pathplanner.lib.PathConstraints;
+import com.pathplanner.lib.PathPlanner;
+import com.pathplanner.lib.PathPlannerTrajectory;
+import com.pathplanner.lib.PathPoint;
+
+import edu.wpi.first.math.geometry.Pose2d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Transform2d;
+import edu.wpi.first.math.geometry.Translation2d;
+import edu.wpi.first.wpilibj2.command.CommandBase;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Constants.Position;
+import frc.robot.subsystems.DriveTrain;
+import frc.robot.subsystems.Vision;
+
+public class TagPositionDrive extends CommandBase {
+    private DriveTrain m_driveTrain;
+    private SequentialCommandGroup m_followTrajectory;
+    private Vision m_vision;
+    private Position m_targetPosition;
+
+    //converted to Transform because cannot add two Poses together
+    private static final Map<Position, Transform2d> ROBOT_POSITIONS = new HashMap<Position, Transform2d>() {
+        {
+            // scoring transformations, change later
+            put(Position.LEFT_TOP, new Transform2d(new Translation2d(3, -4.5), new Rotation2d(0))); // position1
+            put(Position.CENTER_TOP, new Transform2d(new Translation2d(3, 0), new Rotation2d(0)));
+            put(Position.RIGHT_TOP, new Transform2d(new Translation2d(3, 4.5), new Rotation2d(0)));
+            put(Position.LEFT_MIDDLE, new Transform2d(new Translation2d(3, -4.5), new Rotation2d(0)));
+            put(Position.CENTER_MIDDLE, new Transform2d(new Translation2d(3, 0), new Rotation2d(0)));
+            put(Position.RIGHT_MIDDLE, new Transform2d(new Translation2d(3, 4.5), new Rotation2d(0)));
+            put(Position.LEFT_BOTTOM, new Transform2d(new Translation2d(3, -4.5), new Rotation2d(0)));
+            put(Position.CENTER_BOTTOM, new Transform2d(new Translation2d(3, 0), new Rotation2d(0)));
+            put(Position.RIGHT_BOTTOM, new Transform2d(new Translation2d(3, 4.5), new Rotation2d(0)));
+            // substation positions, change later
+            put(Position.LEFT_SUBSTATION, new Transform2d(new Translation2d(100, 0), new Rotation2d(0)));
+            put(Position.RIGHT_SUBSTATION, new Transform2d(new Translation2d(100, 0), new Rotation2d(0)));
+
+        }
+    };
+
+    /** Creates a new ChargeStationDrive. */
+    public TagPositionDrive(DriveTrain driveTrain, Vision vision, Position targetPosition) {
+        this.m_driveTrain = driveTrain;
+        this.m_vision = vision;
+        //pass in a Position ex. pass in Position.LEFT_TOP
+        this.m_targetPosition = targetPosition; 
+        // Use addRequirements() here to declare subsystem dependencies.
+        addRequirements(m_driveTrain);
```

Comment:
I don't think we want this to "require" driveTrain. The actual TrajFollow command will handle that. I suspect that having both do it will cause problems.